diff --git a/cli/src/lib.rs b/cli/src/lib.rs
index 4fdf34b0..0ef43c9e 100644
--- a/cli/src/lib.rs
+++ b/cli/src/lib.rs
@@ -129,7 +129,7 @@ pub enum Command {
         #[clap(long)]
         no_docs: bool,
         /// Architecture to use when building the program
-        #[clap(value_enum, long, default_value = "bpf")]
+        #[clap(value_enum, long, default_value = "sbf")]
         arch: ProgramArch,
     },
     /// Expands macros (wrapper around cargo expand)
@@ -166,7 +166,7 @@ pub enum Command {
         #[clap(value_enum, short, long, default_value = "none")]
         bootstrap: BootstrapMode,
         /// Architecture to use when building the program
-        #[clap(value_enum, long, default_value = "bpf")]
+        #[clap(value_enum, long, default_value = "sbf")]
         arch: ProgramArch,
         /// Environment variables to pass into the docker container
         #[clap(short, long, required = false)]
@@ -199,7 +199,7 @@ pub enum Command {
         #[clap(long)]
         skip_build: bool,
         /// Architecture to use when building the program
-        #[clap(value_enum, long, default_value = "bpf")]
+        #[clap(value_enum, long, default_value = "sbf")]
         arch: ProgramArch,
         /// Flag to keep the local validator running after tests
         /// to be able to check the transactions.
@@ -297,7 +297,7 @@ pub enum Command {
         #[clap(long)]
         skip_build: bool,
         /// Architecture to use when building the program
-        #[clap(value_enum, long, default_value = "bpf")]
+        #[clap(value_enum, long, default_value = "sbf")]
         arch: ProgramArch,
     },
     /// Keypair commands.
@@ -320,7 +320,7 @@ pub enum Command {
         #[clap(long)]
         skip_lint: bool,
         /// Architecture to use when building the program
-        #[clap(value_enum, long, default_value = "bpf")]
+        #[clap(value_enum, long, default_value = "sbf")]
         arch: ProgramArch,
         /// Environment variables to pass into the docker container
         #[clap(short, long, required = false)]
